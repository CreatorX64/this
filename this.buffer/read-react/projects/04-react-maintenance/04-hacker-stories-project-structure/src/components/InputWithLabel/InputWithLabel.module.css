/* If using CSS-in-JS, where no CSS file is needed, one could still have a
separate style.js file for all the styled components */

.label {
  padding-left: 5px;
  font-size: 24px;
}

.input {
  border: none;
  border-bottom: 1px solid #171212;
  font-size: 24px;
  background-color: transparent;
}
