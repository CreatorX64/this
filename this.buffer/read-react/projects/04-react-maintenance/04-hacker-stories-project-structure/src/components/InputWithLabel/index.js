import { useRef, useEffect } from "react";
import styles from "./InputWithLabel.module.css";

const InputWithLabel = ({
  children,
  id,
  value,
  type = "text",
  onInputChange,
  isFocused
}) => {
  const inputRef = useRef();

  useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);

  return (
    <>
      <label htmlFor={id} className={styles.label}>
        {children}
      </label>
      &nbsp;
      <input
        ref={inputRef}
        type={type}
        id={id}
        value={value}
        onChange={onInputChange}
        className={styles.input}
        // autoFocus={isFocused}
      />
    </>
  );
};

export default InputWithLabel;
