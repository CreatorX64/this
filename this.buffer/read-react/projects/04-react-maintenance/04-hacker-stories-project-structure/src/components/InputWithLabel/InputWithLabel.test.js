import { render } from "@testing-library/react";
import InputWithLabel from "./index";

describe("InputWithLabel", () => {
  const inputWithLabelProps = {
    children: "dummy-label-text",
    id: "dummy-id",
    value: "dummy-value"
  };

  test("renders snapshot", () => {
    const onInputChange = jest.fn();
    const { container } = render(
      <InputWithLabel {...inputWithLabelProps} onInputChange={onInputChange} />
    );
    expect(container).toMatchSnapshot();
  });
});
