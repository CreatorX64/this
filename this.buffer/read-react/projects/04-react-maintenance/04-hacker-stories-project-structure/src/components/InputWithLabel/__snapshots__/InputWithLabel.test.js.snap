// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel renders snapshot 1`] = `
<div>
  <label
    class="label"
    for="dummy-id"
  >
    dummy-label-text
  </label>
  Â 
  <input
    class="input"
    id="dummy-id"
    type="text"
    value="dummy-value"
  />
</div>
`;
